src/hera/shared/_global_config.py

```diff
@@ -27,12 +29,12 @@ class _GlobalConfig:
     namespace: Optional[str] = None
     _image: Union[str, Callable[[], str]] = "python:3.7"
     service_account_name: Optional[str] = None
-    script_command: Optional[List[str]] = ["python"]
+    script_command: Optional[List[str]] = field(default_factory=lambda: ["python"])
     _pre_build_hooks: Optional[_HookMap] = None
 
     def reset(self) -> None:
         """Resets the global config container to its initial state"""
-        self.__dict__.clear()  # Wipe instance values to fallback to the class defaults
+        self.__dict__ = _GlobalConfig().__dict__
```

Comment:
Creating a new instance and sourcing values from there to ensure it sets it to defaults.